main/tests/test_views.py

```diff
@@ -0,0 +1,128 @@
+        """Prepare data for testing."""
+        self.client = Client()
+        self.admin = User.objects.create(username='testuser', password='blablabla', is_superuser=True, is_staff=True,
+                                         email='testuser@gmail.com', is_active=True)
+
+    def test_Index_view(self):
+        """Testing main page."""
+        post = Post.objects.create(author=self.admin, title='Test', text='superText', created_date=timezone.now())
+        post.publish()
+        response = self.client.get('/')
+        self.assertEqual(200, response.status_code)
+        self.assertTemplateUsed(response, 'main/index.html')
+        self.assertContains(response, 'Test')
+
+    def test_detail_view(self):
+        """Testing detail page when post is not exist and when it exists."""
+        response = self.client.get('/post/1/')
+        self.assertEqual(404, response.status_code)
+        Post.objects.create(author=self.admin, title='Test', text='superText', created_date=timezone.now())
+        response = self.client.get('/post/1/')
+        self.assertEqual(200, response.status_code)
+
+    def test_post_new_view(self):
+        """Testing new post view before and after login."""
+        response = self.client.get('/post/new/')
+        self.assertEqual(302, response.status_code)
+        authorization = self.client.login(username='admin', password='12345678')
+        self.assertTrue(authorization)
+        response = self.client.post('/post/new/', {'author': self.admin, 'title': 'Test', 'text': 'superText',
+                                                   'created_date': timezone.now()}, follow=True)
+        self.assertEqual(200, response.status_code)
+        # Here's we have redirect_chain[0][0] because it's return list of tuples. We had only one redirect
+        # so our list has only 1 element, tuple has 2 - first it's a link where we were redirected and second
+        # it's a status code
+        self.assertEqual('/post/1/', response.redirect_chain[0][0])
+
+    def test_post_edit(self):
+        """Testing edit views before and after user login."""
+        response = self.client.get('/post/1/edit/')
+        self.assertEqual(302, response.status_code)
+        authorization = self.client.login(username='admin', password='12345678')
+        self.assertTrue(authorization)
+        response = self.client.get('/post/1/edit/')
+        # This time we'are logged but post is not exist
+        self.assertEqual(404, response.status_code)
+        Post.objects.create(author=self.admin, title='Test', text='superText', created_date=timezone.now())
+        response = self.client.get('/post/1/edit/')
+        self.assertEqual(200, response.status_code)
+
+    def test_drafts(self):
+        """Testing drafts view before and after login."""
+        response = self.client.get('/drafts/')
+        self.assertEqual(302, response.status_code)
+        authorization = self.client.login(username='admin', password='12345678')
```

Comment:
You many times use `username='admin'` and  `password='12345678'`, better way is create constants in test module or test class like `USERNAME='admin'`, `PASSWORD='12345678'`